### ISSUE TITLE
string comparison with `sys.version` (python3.10)

### ISSUE THREAD

**Issue by asottile** [asottile/flake8-2020#4]
I haven't seen this one in the wild, but I think it's ~reasonably well defined for 3.0 - 3.9 so it's possible:

```pycon
Python 3.5.7 (default, Mar 30 2019, 01:05:18) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.version >= '3.5'
True
```

```pycon
Python 3.10.0a0 (default, Jul 18 2019, 06:46:14) 
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.version >= '3.5'
False
```

**Comment by hugovk** [comment]
Hi, looks like #5 has a false positive in for single character comparisons. Running on scipy:

```console
$ flake8 --select YTT scipy
scipy/_lib/decorator.py:45:4: YTT103: `sys.version` compared to string (python3.10), use `sys.version_info`
scipy/_lib/decorator.py:113:20: YTT103: `sys.version` compared to string (python3.10), use `sys.version_info`
```

[First one](http://github.com/scipy/scipy/blob/1465ab6f40650ae66bab40e8db861aec757a83a6/scipy/_lib/decorator.py#L45-L46):

```python
if sys.version >= '3':
    from inspect import getfullargspec
```

[Second one](https://github.com/scipy/scipy/blob/1465ab6f40650ae66bab40e8db861aec757a83a6/scipy/_lib/decorator.py#L113-L117):

```python
                if sys.version < '3':  # easy way
                    self.shortsignature = self.signature = (
                        inspect.formatargspec(
                            formatvalue=lambda val: "", *argspec)[1:-1])
                else:  # Python 3 way
```

Simplified passing examples:

```python
v2_7 = "2.7.16 (default, Mar  4 2019, 09:02:22) \n[GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.11.45.5)]"
assert v2_7 < "3"
assert v2_7 >= "2"
assert v2_7 > "1"

v3_7 = "3.7.4 (default, Jul  9 2019, 18:13:23) \n[Clang 10.0.1 (clang-1001.0.46.4)]"
assert v3_7 < "4"
assert v3_7 >= "3"
assert v3_7 > "2"

v3_10 = "3.10.4 (default, Jul  9 2019, 18:13:23) \n[Clang 10.0.1 (clang-1001.0.46.4)]"
assert v3_10 < "4"
assert v3_10 >= "3"
assert v3_10 > "2"

v4 = "4.0.4 (default, Jul  9 2019, 18:13:23) \n[Clang 10.0.1 (clang-1001.0.46.4)]"
assert v4 < "5"
assert v4 >= "4"
assert v4 > "3"
```
---

Having said that, these string comparisons aren't very safe (although Python 10 and 20 are some time off!): 

```python
v20 = "20.0.4 (default, Jul  9 2019, 18:13:23) \n[Clang 10.0.1 (clang-1001.0.46.4)]"
assert v20 < "3"  # SHOULD NOT PASS!
assert v20 >= "3"  # SHOULD PASS!
```

**Comment by hugovk** [comment]
Yeah, although the warning says `python3.10`, not `python10`.
